lms/djangoapps/ora_staff_grader/serializers.py

```diff
@@ -196,6 +197,22 @@ class UploadedFileSerializer(serializers.Serializer):
     name = serializers.CharField()
     size = serializers.IntegerField()
 
+    def to_representation(self, instance):
```

Comment:
Would it be possible instead to change the serializer of downloadURL? I'm thinking something along the lines of: `downloadUrl = AbsoluteURLField(source="download_url")`

Not sure if my proposition is better, or even possible at all.

Reply:
Do you mean something like this?
```
class AbsoluteURLField(serializers.URLField):
    def to_representation(self, value):
        if value and not value.startswith(("http://", "https://")):
            return f"{settings.LMS_ROOT_URL}{value}"
        return value

class UploadedFileSerializer(serializers.Serializer):
    downloadUrl = AbsoluteURLField(source="download_url")
    ...
```

Reply:
I wonder if you considered using SerializerMethodField instead of overriding `to_representation`. I tried this and it worked:
```
class UploadedFileSerializer(serializers.Serializer):
    """Serializer for a file uploaded as a part of a response"""

    downloadUrl = serializers.SerializerMethodField(method_name="get_download_url")
    description = serializers.CharField()
    name = serializers.CharField()
    size = serializers.IntegerField()

    def get_download_url(self, obj):
        """
        Get the representation for SerializerMethodField `downloadUrl`
        """
        return urljoin(settings.LMS_ROOT_URL, obj["download_url"])
```
So, I'm curious about why this is the chosen approach. Thanks!